ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -197,33 +209,40 @@ private[commands] class CommandTracker[Context](
 
       import CommandTracker.nonTerminalCodes
 
-      private def handleSubmitResponse(submitResponse: Ctx[(Context, String), Try[Empty]]) = {
-        val Ctx((_, commandId), value, _) = submitResponse
+      private def handleSubmitResponse(
+          submitResponse: Ctx[(Context, TrackedCommandKey), Try[Empty]]
+      ) = {
+        val Ctx((_, commandKey), value, _) = submitResponse
         value match {
           case Failure(GrpcException(status @ GrpcStatus(code, _), metadata))
               if !nonTerminalCodes(code) =>
-            getOutputForTerminalStatusCode(commandId, GrpcStatus.toProto(status, metadata))
+            getOutputForTerminalStatusCode(commandKey, GrpcStatus.toProto(status, metadata))
           case Failure(throwable) =>
             logger.warn(
               s"Service responded with error for submitting command with context ${submitResponse.context}. Status of command is unknown. watching for completion...",
               throwable,
             )
             None
           case Success(_) =>
-            logger.trace("Received confirmation that command {} was accepted.", commandId)
+            logger.trace(
+              "Received confirmation that command {} from submission {} was accepted.",
+              commandKey.commandId,
+              commandKey.submissionId,
+            )
             None
         }
       }
 
       @nowarn("msg=deprecated")
       private def registerSubmission(submission: Ctx[Context, CommandSubmission]): Unit = {
         val commands = submission.value.commands
+        val submissionId = commands.submissionId
         val commandId = commands.commandId
-        logger.trace("Begin tracking of command {}", commandId)
-        if (pendingCommands.contains(commandId)) {
+        logger.trace("Begin tracking of command {} for submission {}", commandId, submissionId)
+        if (pendingCommands.contains(TrackedCommandKey(submissionId, commandId))) {
           // TODO return an error identical to the server side duplicate command error once that's defined.
           throw new IllegalStateException(
-            s"A command with id $commandId is already being tracked. CommandIds submitted to the CommandTracker must be unique."
+            s"A command $commandId from a submission $submissionId is already being tracked. CommandIds submitted to the CommandTracker must be unique."
           ) with NoStackTrace
```

Comment:
Do you think we have any gaps in testing though for this? Wondering how come conformance tests were green considering this code. Shouldn't they throw an error for 2 submissions with the same command id currently? 

Also that TODO, should we throw an ApiException for duplicate command from here?

Reply:
Thanks for the feedback!

>Do you think we have any gaps in testing though for this? Wondering how come conformance tests were green considering this code. Shouldn't they throw an error for 2 submissions with the same command id currently?

Not yet sure, I was going to think about this tomorrow, as I'm also not sure about some of these changes at all. Partially because I don't know Akka Streams Graph API 😄 I submitted it mostly as an FYI so far.

> Also that TODO, should we throw an ApiException for duplicate command from here?
 
Was also thinking about this. Will keep you posted.

Reply:
Sounds good, let me know if you need any help with the Graph API, spent a lot of time writing akka graphs.

Reply:
> Also that TODO, should we throw an ApiException for duplicate command from here?

Not sure what to do, as this is a `Unit` method. Maybe let's keep it as is for now and I can do something about this in a separate PR.